fuse/file_test.go

```diff
@@ -0,0 +1,653 @@
+package fuse
+
+import (
+	"context"
+	"testing"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+	"github.com/puppetlabs/wash/datastore"
+	"github.com/puppetlabs/wash/plugin"
+	plugintest "github.com/puppetlabs/wash/plugin/test"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+type fileTestSuite struct {
+	suite.Suite
+	ctx context.Context
+}
+
+func (suite *fileTestSuite) SetupTest() {
+	plugin.SetTestCache(datastore.NewMemCache())
+}
+
+func (suite *fileTestSuite) TearDownTest() {
+	plugin.UnsetTestCache()
+}
+
+func (suite *fileTestSuite) TestOpen_File() {
```

Comment:
Nit: `File` => `FileLikeEntry` ? Latter's more precise and matches what's in the code.

Reply:
And `TestOpen_NonFile` to `TestOpen_NonFileLikeEntry`? FileLikeEntry is a mouthful, I was using an obvious short-hand.

Reply:
Doesn't look too bad, but it is a nit so am OK if you want to leave it as-is.

Reply:
Actually, it does help make the other tests a bit easier to read for me. Like `Read_File` is better understood as `Read_FileLikeEntry`. So would be nice to make that change.

Reply:
It'd also help to name this test `TestOpen_FileLikeEntry_OpenFlags` since you're testing that Open does the right thing for all combos of openflag and entry. Similar case for the other one.

Reply:
Done.